workbench-core/base/src/aws/helpers/serviceCatalogService.ts

```diff
@@ -38,4 +47,56 @@ export default class ServiceCatalogService {
 
     return portfolio ? portfolio.Id : undefined;
   }
+
+  /**
+   * Get portfolio products
+   * @param portfolioId - Id of the portfolio
+   */
+  public async getProductsByPortfolioId(portfolioId: string): Promise<ProductViewSummary[]> {
+    let productsDetails: ProductViewDetail[] = [];
+    let pageToken: string | undefined = undefined;
+    // Get all products within portfolio
+    do {
+      const productsFilter: SearchProductsAsAdminCommandInput = {
+        PageToken: pageToken,
+        PageSize: 20,
+        SortBy: 'CreationDate',
+        SortOrder: 'DESCENDING',
+        PortfolioId: portfolioId
+      };
+      const productList = await this._serviceCatalog.searchProductsAsAdmin(productsFilter);
+      pageToken = productList.NextPageToken;
+      if (productList.ProductViewDetails) {
+        productsDetails = productsDetails.concat(productList.ProductViewDetails);
+      }
+    } while (pageToken);
```

Comment:
the lambda running this code has a timeout of 2 minutes to avoid multiple lambdas running together, so If in any case it takes too long it wont cause further issues than not adding new environment types, and about the pagination, thats the reason we are filtering by portfolio , we are not expecting a customer to have too many products to make this take more than 2 minutes, but the purpose of this process is to scan all products inside a portfolio and if there is a new one found we add a new env type.

Reply:
It's actually 4 minutes right?
https://github.com/aws-solutions/solution-spark-on-aws/blob/develop/solutions/swb-reference/src/SWBStack.ts#L527

Reply:
Out of curiosity, how long did it take the lambda to run in your account with the new envType code change?

Reply:
@FernandoAranda makes sense. It may be beyond the scope of this feature, but if we're just looking for new items, it might be easier to use SNS so we can fire an event that there is a new product when it gets created instead of having to search through all of them.

Reply:
> Out of curiosity, how long did it take the lambda to run in your account with the new envType code change?

in the logs its only 4 seconds

Reply:
> @FernandoAranda makes sense. It may be beyond the scope of this feature, but if we're just looking for new items, it might be easier to use SNS so we can fire an event that there is a new product when it gets created instead of having to search through all of them.
> 
> I'm fine with this being resolved once @nguyen102's questions are resolved.

yeah that would make things easier in the validation side but add dev time for the SNS , for now the cron job makes sense as it already exists and runs every 5 minutes